alacritty/src/display.rs

```diff
@@ -216,41 +213,30 @@ impl Display {
         let (glyph_cache, cell_width, cell_height) =
             Self::new_glyph_cache(dpr, &mut renderer, config)?;
 
-        let padding = config.ui_config.window.padding;
-        let mut padding_x = f32::from(padding.x) * dpr as f32;
-        let mut padding_y = f32::from(padding.y) * dpr as f32;
-
-        if let Some((width, height)) =
-            GlyphCache::calculate_dimensions(&config.ui_config.window, dpr, cell_width, cell_height)
-        {
-            let PhysicalSize { width: w, height: h } = window.inner_size();
-            if w == width && h == height {
+        if let Some(dimensions) = dimensions {
+            if (estimated_dpr - dpr).abs() < f64::EPSILON {
                 info!("Estimated DPR correctly, skipping resize");
             } else {
-                window.set_inner_size(PhysicalSize::new(width, height));
+                // Resize the window again if the DPR was not estimated correctly.
+                let size = window_size(config, dimensions, cell_width, cell_height, dpr);
+                window.set_inner_size(size);
             }
-        } else if config.ui_config.window.dynamic_padding {
-            // Make sure additional padding is spread evenly.
-            padding_x = dynamic_padding(padding_x, viewport_size.width as f32, cell_width);
-            padding_y = dynamic_padding(padding_y, viewport_size.height as f32, cell_height);
         }
 
-        padding_x = padding_x.floor();
-        padding_y = padding_y.floor();
-
-        info!("Cell Size: {} x {}", cell_width, cell_height);
-        info!("Padding: {} x {}", padding_x, padding_y);
-
         // Create new size with at least one column and row.
-        let size_info = SizeInfo {
-            dpr,
-            width: (viewport_size.width as f32).max(cell_width + 2. * padding_x),
-            height: (viewport_size.height as f32).max(cell_height + 2. * padding_y),
+        let size_info = SizeInfo::new(
+            viewport_size.width as f32,
+            viewport_size.height as f32,
             cell_width,
             cell_height,
-            padding_x,
-            padding_y,
-        };
+            f32::from(config.ui_config.window.padding.x),
+            f32::from(config.ui_config.window.padding.y),
+            dpr,
+            config.ui_config.window.dynamic_padding,
+        );
+
+        info!("Cell Size: {} x {}", cell_width, cell_height);
```

Comment:
Just to be consistent with other renames.

```suggestion
        info!("Cell size: {} x {}", cell_width, cell_height);
```